### PR TITLE
Modify DumpClusterLogs in gke deployer to let log-dump.sh use gcloud scp

### PR DESCRIPTION
**Pull request by shyamjvs**
This complements changes on k8s-side in PR https://github.com/kubernetes/kubernetes/pull/51834
Ref issue #4323 

/cc @krzyzacy

### CODE REVIEW

**Comment by shyamjvs** [comment]
@kubernetes/test-infra-reviewers Could someone review this? It's important for debugging gke tests.

File: kubetest/gke.go [hunk 1]
```diff
@@ -66,7 +65,9 @@ type gkeNodePool struct {
 
 type gkeDeployer struct {
 	project         string
-	location        string
+	zone            string
+	region          string
+	locationArg     string
```
Comment (wojtek-t) [inline kubetest/gke.go]: Why do you name it locationArg? Why not leave location?

File: kubetest/gke.go [hunk 2]
```diff
@@ -287,23 +290,28 @@ func (g *gkeDeployer) IsUp() error {
 //
 // TODO(zmerlynn): This whole path is really gross, but this seemed
 // the least gross hack to get this done.
+//
+// TODO(shyamjvs): Make this work with regional clusters where
```
Comment (wojtek-t) [inline kubetest/gke.go]: Is it only about regional clusters?
I think it's also about multizonal clusters (where you have single master, but nodes in different zones).

File: kubetest/gke.go (hunk 1 above)
Comment (shyamjvs) [reply kubetest/gke.go]: Changed it. Thought that was more apt as it's not the value of the location exactly, but the argument (i.e --zone=foobar)

File: kubetest/gke.go (hunk 2 above)
Comment (shyamjvs) [reply kubetest/gke.go]: Changed the comment. Though I'm not sure if we're running any gke tests that are multizonal except the regional one.
Federation has some tests on gce with clusters from different zones, but not on gke.

**Comment by shyamjvs** [comment]
@wojtek-t Addressed comments. PTAL

File: kubetest/gke.go [hunk 3]
```diff
@@ -287,23 +290,27 @@ func (g *gkeDeployer) IsUp() error {
 //
 // TODO(zmerlynn): This whole path is really gross, but this seemed
 // the least gross hack to get this done.
+//
+// TODO(shyamjvs): Make this work with multizonal clusters.
```
Comment (wojtek-t) [inline kubetest/gke.go]: and regional

Comment (shyamjvs) [reply kubetest/gke.go]: Done. Btw - isn't regional a subset of multi-zonal (where all zones are from same region)?

**Comment by shyamjvs** [comment]
@wojtek-t PTAL

File: kubetest/gke.go (hunk 3 above)
Comment (wojtek-t) [reply kubetest/gke.go]: multi-zonal is always within a single region.

It doesn't have to be a subset - you can have a regional GKE cluster, where all nodes will be in a single zone.

**Pull request merged**